DokaSnake/DokaSnake/DokaSnake.cpp

```diff
@@ -48,18 +50,46 @@ class Map {
 };
 
 
-void HeroInit() {
-	SetXY(5, 10);
+void HeroSummon(int x, int y) {
+	SetXY(x, y);
+	X = x;
+	Y = y;
 	cout << "@";
+	Sleep(100);
+}
+
+void Erease(int x, int y) {
+	SetXY(x, y);
+	cout << " ";
 }
 
 int main()
 {
+	bool death = 0;
 	Map Map;
 	Map.MapPreSet();
 	Map.MapBuilder();
-	while (1) {
-		HeroInit();
+	HeroSummon(X, Y);
+	while (!death) {
+		switch (_getch()) {
+		case RIGHT:
+			Erease(X, Y);
+			HeroSummon(X, Y + 1);
+			break;
+		case UP:
+			Erease(X, Y);
+			HeroSummon(X - 1, Y);
+			break;
+		case LEFT:
+			Erease(X, Y);
+			HeroSummon(X, Y - 1);
+			break;
+		case DOWN:
+			Erease(X, Y);
+			HeroSummon(X + 1, Y);
+			break;
+		}
```

Comment:
In all that cases you need to check presence of walls
For example, in right case you need to check wall at (x, y + 1), if it exist, so don't move hero, otherwise you can summon  hero